src/ini.cr

```diff
@@ -19,4 +19,21 @@ class INI
     end
     ini
   end
+
+  # Generates an INI-style configuration from a given hash.
+  #
+  # ```
+  # INI.build({"foo" => {"a" => "1"}}, " ") # => "[foo]\na = 1\n\n"
+  # ```
+  def self.build(ini : Hash(String, Hash(String, String)), space : String = "") : String
+    io = IO::Memory.new
```

Comment:
You should use a `String::Builder` instead of `Memory::IO`, because this is optimized for creating a string.
And please add a method overload `INI.build(io : IO, hash)` that can directly write to an IO instead of allocating a string.

Reply:
I've first found `IO::Memory.new` if the [Crystal performance guide](https://crystal-lang.org/docs/guides/performance.html). So we can further improve perfomance using `String::Builder` for strings; I will make a PR for this, thanks.

Reply:
I've not found yet a case when `IO::Memory.new` is slower than the `String.build` approach, using `Benchmark.ips`. Anyone has a sample to demonstrate this? We will be able to add it to the Performance Guide.

Reply:
```crystal
require "benchmark"

Benchmark.ips do |x|
  x.report("String.build") do
    String.build do |io|
      100.times do
        io << "hello world"
      end
    end
  end
  x.report("IO::Memory") do
    io = IO::Memory.new
    100.times do
      io << "hello world"
    end
    io.to_s
  end
end
```

Output:

```
String.build 475.55k (   2.1µs) (± 1.25%)       fastest
  IO::Memory  252.7k (  3.96µs) (± 1.47%)  1.88× slower
```

whats your benchmark?

Reply:
@larubujo Thanks! I haven't used `100.times` before. I had just added a block similar to ones already present in the [guide](https://crystal-lang.org/docs/guides/performance.html). So I've faced strange issues, memory related I think.